sdk/dotnet/Pulumi.Automation/Commands/IPulumiCmd.cs

```diff
@@ -11,7 +11,7 @@ namespace Pulumi.Automation.Commands
     internal interface IPulumiCmd
     {
         Task<CommandResult> RunAsync(
-            IEnumerable<string> args,
+            IList<string> args,
```

Comment:
Hm, this was an automatically suggested change? I wonder why it did that. It seems to me IEnumerable has fewer assumptions than IList, and if RunAsync does not need random access, the change weakens it? Just curious why it decided to change this.

Reply:
No, the issue was were multiple enumerations of an enumerable in some places. I then looked carefully at caller usage and determined that it was better to fix the warning by promoting `args` from `IEnumerable` to `IList` and doing so consistently.